test/test_sparse.py

```diff
@@ -166,6 +166,20 @@ def f5():
             self.assertEqual(len(cm.warnings), 1)
 
 
+    def test_legacy_ctor_validates(self):
+
+        with self.assertRaisesRegex(
+            RuntimeError,
+            "size is inconsistent with indices: for dim 0, size is 1 but found index 4702111234474983745"
+        ):
+            x = torch.sparse.FloatTensor(
+                torch.tensor([[4702111234474983745], [0]]),
```

Comment:
nit: you can use 3 as the index here to avoid the weirdly large value. It should fail the same way !